From 8652100545c84c1db5f59840ba34c08615f1e572 Mon Sep 17 00:00:00 2001
From: MeiChia Chiu <meichia.chiu@mediatek.com>
Date: Tue, 1 Aug 2023 16:02:28 +0800
Subject: [PATCH 1021/1044] mtk: wifi: mt76: mt7996: Establish BA in VO queue

---
 mt7996/mac.c | 2 --
 1 file changed, 2 deletions(-)

--- a/mt7996/mac.c
+++ b/mt7996/mac.c
@@ -1031,8 +1031,6 @@ mt7996_tx_check_aggr(struct ieee80211_st
 		return;
 
 	tid = skb->priority & IEEE80211_QOS_CTL_TID_MASK;
-	if (tid >= 6) /* skip VO queue */
-		return;
 
 	if (is_8023) {
 		fc = IEEE80211_FTYPE_DATA |
