--- a/mt7915/mcu.c
+++ b/mt7915/mcu.c
@@ -2905,7 +2905,7 @@ static int mt7915_mcu_set_pre_cal(struct
 	skb_put_data(skb, &req, sizeof(req));
 	skb_put_data(skb, data, len);
 
-	return mt76_mcu_skb_send_msg(&dev->mt76, skb, cmd, false);
+	return mt76_mcu_skb_send_msg(&dev->mt76, skb, cmd, true);
 }
 
 int mt7915_mcu_apply_group_cal(struct mt7915_dev *dev)
@@ -2926,6 +2926,9 @@ int mt7915_mcu_apply_group_cal(struct mt
 
 		len = min_t(u32, total, MT_EE_CAL_UNIT);
 
+		dev_dbg(dev->mt76.dev, "apply pre-cal idx=%u cal[0]=0x%02x len=%d total=%d\n",
+				idx, ((unsigned char *)cal)[0], len, total);
+
 		ret = mt7915_mcu_set_pre_cal(dev, idx, cal, len,
 					     MCU_EXT_CMD(GROUP_PRE_CAL_INFO));
 		if (ret)
