--- a/mt7996/eeprom.c
+++ b/mt7996/eeprom.c
@@ -474,7 +474,13 @@ static int mt7996_eeprom_load_precal(str
 
 	mt7996_eeprom_init_precal(dev);
 
-	if (dev->eeprom_mode != EEPROM_MODE_FLASH || !val)
+	if (dev->eeprom_mode != EEPROM_MODE_FLASH)
+		return 0;
+
+	/* if only bit 6 is set for OOB emission workaround there's no
+	 * need to allocate and load pre-cal data from flash
+	 */
+	if (!(val & MT_EE_WIFI_CAL_GROUP) && !(val & MT_EE_WIFI_CAL_DPD))
 		return 0;
 
 	size = MT_EE_CAL_GROUP_SIZE + MT_EE_CAL_DPD_SIZE;
@@ -490,7 +496,15 @@ static int mt7996_eeprom_load_precal(str
 	if (!ret)
  		return ret;
 
-	return mt76_get_of_data_from_nvmem(mdev, dev->cal, "precal", size);
+	ret = mt76_get_of_data_from_nvmem(mdev, dev->cal, "precal", size);
+	if (!ret)
+ 		return ret;
+
+	dev_warn(mdev->dev, "missing precal data, size=%d\n", size);
+	devm_kfree(mdev->dev, dev->cal);
+	dev->cal = NULL;
+
+	return ret;
 }
 
 int mt7996_eeprom_init(struct mt7996_dev *dev)
