--- a/mt7915/mac.c
+++ b/mt7915/mac.c
@@ -1276,9 +1276,10 @@ static void
 mt7915_mac_full_reset(struct mt7915_dev *dev)
 {
 	struct mt76_phy *ext_phy;
-	struct mt7915_phy *phy2;
+	struct mt7915_phy *phy, *phy2;
 	int i;
 
+	phy = &dev->phy;
 	ext_phy = dev->mt76.phys[MT_BAND1];
 	phy2 = ext_phy ? ext_phy->priv : NULL;
 
@@ -1316,6 +1317,10 @@ mt7915_mac_full_reset(struct mt7915_dev
 	INIT_LIST_HEAD(&dev->sta_rc_list);
 	INIT_LIST_HEAD(&dev->twt_list);
 
+	INIT_LIST_HEAD(&phy->stats_list);
+	if (phy2)
+		INIT_LIST_HEAD(&phy2->stats_list);
+
 	i = mt76_wcid_alloc(dev->mt76.wcid_mask, MT7915_WTBL_STA);
 	dev->mt76.global_wcid.idx = i;
 	dev->recovery.hw_full_reset = false;
