--- a/mt7915/init.c
+++ b/mt7915/init.c
@@ -6,6 +6,7 @@
 #include <linux/hwmon-sysfs.h>
 #include <linux/of.h>
 #include <linux/thermal.h>
+#include <linux/of.h>
 #include "mt7915.h"
 #include "mac.h"
 #include "mcu.h"
@@ -867,6 +868,11 @@ mt7915_init_hardware(struct mt7915_dev *
 
 	INIT_WORK(&dev->init_work, mt7915_init_work);
 
+	if (dev->mt76.dev->of_node) {
+		of_property_read_u32(dev->mt76.dev->of_node, "rdd_antenna",
+		                     &dev->phy.rdd_antenna);
+	}
+
 	ret = mt7915_dma_init(dev, phy2);
 	if (ret)
 		return ret;
--- a/mt7915/mac.c
+++ b/mt7915/mac.c
@@ -1973,6 +1973,27 @@ static void mt7915_dfs_stop_radar_detect
 	mt76_connac_mcu_rdd_cmd(&dev->mt76, RDD_STOP, rdd_idx, 0, 0);
 }
 
+static int mt7915_dfs_set_wf_ant(struct mt7915_dev *dev, int rdd_idx, int start)
+{
+	int err = 0;
+
+	if (is_mt7915(&dev->mt76)) {
+		u32 rdd_antenna = dev->dbdc_support ? 2 : 0;
+
+		if (dev->phy.rdd_antenna)
+			rdd_antenna = dev->phy.rdd_antenna;
+
+		if (start)
+			dev_info(dev->mt76.dev, "%s: dts rdd_antenna=%u rdd_antenna=%u rdd_idx=%d\n",
+			         __func__, dev->phy.rdd_antenna, rdd_antenna, rdd_idx);
+
+		err = mt76_connac_mcu_rdd_cmd(&dev->mt76, RDD_SET_WF_ANT, rdd_idx,
+		                              0, rdd_antenna);
+	}
+
+	return err;
+}
+
 static int mt7915_dfs_start_rdd(struct mt7915_dev *dev, int rdd_idx)
 {
 	int err, region;
@@ -1994,12 +2015,18 @@ static int mt7915_dfs_start_rdd(struct m
 	if (err < 0)
 		return err;
 
+#if 0
 	if (is_mt7915(&dev->mt76)) {
 		err = mt76_connac_mcu_rdd_cmd(&dev->mt76, RDD_SET_WF_ANT, rdd_idx,
 					      0, dev->dbdc_support ? 2 : 0);
 		if (err < 0)
 			return err;
 	}
+#endif
+
+	err = mt7915_dfs_set_wf_ant(dev, rdd_idx, 1);
+	if (err < 0)
+		return err;
 
 	return mt76_connac_mcu_rdd_cmd(&dev->mt76, RDD_DET_MODE, rdd_idx, 0, 1);
 }
@@ -2106,13 +2133,16 @@ stop:
 	if (err < 0)
 		return err;
 
+#if 0
 	if (is_mt7915(&dev->mt76)) {
 		err = mt76_connac_mcu_rdd_cmd(&dev->mt76, RDD_SET_WF_ANT,
 					      rdd_idx, 0, dev->dbdc_support ? 2 : 0);
 		if (err < 0)
 			return err;
 	}
+#endif
 
+	mt7915_dfs_set_wf_ant(dev, phy->mt76->band_idx, 0);
 	mt7915_dfs_stop_radar_detector(phy);
 	phy->mt76->dfs_state = MT_DFS_STATE_DISABLED;
 
--- a/mt7915/mt7915.h
+++ b/mt7915/mt7915.h
@@ -229,6 +229,7 @@ struct mt7915_phy {
 	u32 throttle_temp[2]; /* 0: critical high, 1: maximum */
 
 	u32 rxfilter;
+	u32 rdd_antenna;
 	u64 omac_mask;
 
 	u16 noise;
