--- a/mt7996/mac.c
+++ b/mt7996/mac.c
@@ -2359,7 +2359,7 @@ void mt7996_mac_reset_work(struct work_s
 	if (!(READ_ONCE(dev->recovery.state) & MT_MCU_CMD_STOP_DMA))
 		return;
 
-	dev_info(dev->mt76.dev,"\n%s L1 SER recovery start.",
+	dev_info(dev->mt76.dev,"==== %s L1 SER recovery start ====",
 		 wiphy_name(hw->wiphy));
 
 	if (mtk_wed_device_active(&dev->mt76.mmio.wed_hif2))
@@ -2371,6 +2371,9 @@ void mt7996_mac_reset_work(struct work_s
 	mt7996_npu_hw_stop(dev);
 	ieee80211_stop_queues(mt76_hw(dev));
 
+	dev_info(dev->mt76.dev,"%s L1 SER queue stop done",
+		wiphy_name(dev->mt76.hw->wiphy));
+
 	set_bit(MT76_RESET, &dev->mphy.state);
 	set_bit(MT76_MCU_RESET, &dev->mphy.state);
 	mt76_abort_scan(&dev->mt76);
@@ -2384,6 +2387,10 @@ void mt7996_mac_reset_work(struct work_s
 	}
 
 	mt76_worker_disable(&dev->mt76.tx_worker);
+
+	dev_info(dev->mt76.dev,"%s L1 SER disable tx_work done",
+		wiphy_name(dev->mt76.hw->wiphy));
+
 	mt76_for_each_q_rx(&dev->mt76, i) {
 		if (mtk_wed_device_active(&dev->mt76.mmio.wed) &&
 		    mt76_queue_is_wed_rro(&dev->mt76.q_rx[i]))
@@ -2402,14 +2409,30 @@ void mt7996_mac_reset_work(struct work_s
 
 	mutex_lock(&dev->mt76.mutex);
 
+	dev_info(dev->mt76.dev,"%s L1 SER napi disable done",
+		 wiphy_name(dev->mt76.hw->wiphy));
+
 	mt76_wr(dev, MT_MCU_INT_EVENT, MT_MCU_INT_EVENT_DMA_STOPPED);
 
+	dev_info(dev->mt76.dev,"%s L1 SER dma stop done",
+		 wiphy_name(dev->mt76.hw->wiphy));
+
 	if (mt7996_wait_reset_state(dev, MT_MCU_CMD_RESET_DONE)) {
 		mt7996_dma_reset(dev, false);
 
+		dev_info(dev->mt76.dev,"%s L1 SER dma reset done",
+			wiphy_name(dev->mt76.hw->wiphy));
+
 		mt7996_tx_token_put(dev);
+
+		dev_info(dev->mt76.dev,"%s L1 SER token put done",
+			wiphy_name(dev->mt76.hw->wiphy));
+
 		idr_init(&dev->mt76.token);
 
+		dev_info(dev->mt76.dev,"%s L1 SER idr init done",
+			wiphy_name(dev->mt76.hw->wiphy));
+
 		mt76_wr(dev, MT_MCU_INT_EVENT, MT_MCU_INT_EVENT_DMA_INIT);
 		mt7996_wait_reset_state(dev, MT_MCU_CMD_RECOVERY_DONE);
 	}
@@ -2442,6 +2465,9 @@ void mt7996_mac_reset_work(struct work_s
 
 	__mt7996_npu_hw_init(dev);
 
+	dev_info(dev->mt76.dev,"%s L1 SER dma start done",
+		wiphy_name(dev->mt76.hw->wiphy));
+
 	clear_bit(MT76_MCU_RESET, &dev->mphy.state);
 	mt7996_for_each_phy(dev, phy)
 		clear_bit(MT76_RESET, &phy->mt76->state);
@@ -2481,7 +2507,7 @@ void mt7996_mac_reset_work(struct work_s
 	mt7996_for_each_phy(dev, phy)
 		ieee80211_queue_delayed_work(hw, &phy->mt76->mac_work,
 					     MT7996_WATCHDOG_TIME);
-	dev_info(dev->mt76.dev,"\n%s L1 SER recovery completed.",
+	dev_info(dev->mt76.dev,"==== %s L1 SER recovery completed ====",
 		 wiphy_name(dev->mt76.hw->wiphy));
 }
 
@@ -2556,6 +2582,9 @@ skip_coredump:
 
 void mt7996_reset(struct mt7996_dev *dev)
 {
+	dev_info(dev->mt76.dev, "%s SER recovery state: 0x%08x\n",
+		wiphy_name(dev->mt76.hw->wiphy), READ_ONCE(dev->recovery.state));
+
 	if (!dev->recovery.hw_init_done)
 		return;
 
