--- a/mt7615/mcu.c
+++ b/mt7615/mcu.c
@@ -386,7 +386,10 @@ mt7615_mcu_rx_radar_detected(struct mt76
 
 	if (!dev->radar_pattern.n_pulses && !r->long_detected &&
 	    !r->constant_prf_detected && !r->staggered_prf_detected)
+	{
+		dev_info(dev->mt76.dev, "%s: called but no radar detected\n", __func__);
 		return;
+	}
 
 	if (r->band_idx && dev->mt76.phys[MT_BAND1])
 		mphy = dev->mt76.phys[MT_BAND1];
@@ -394,6 +397,12 @@ mt7615_mcu_rx_radar_detected(struct mt76
 	if (mt76_phy_dfs_state(mphy) < MT_DFS_STATE_CAC)
 		return;
 
+	dev_info(dev->mt76.dev, "%s: RADAR DETECTED r->band_idx=%d\n", __func__, r->band_idx);
+	dev_dbg(dev->mt76.dev, "%s: n_pulses=%u long_detected=%u constant_prf_detected=%u staggered_prf_detected=%u\n",
+			__func__, dev->radar_pattern.n_pulses, r->long_detected, r->constant_prf_detected, r->staggered_prf_detected);
+	dev_dbg(dev->mt76.dev, "%s: radar_type_idx=%u periodic_pulse_num=%u long_pulse_num=%u hw_pulse_num=%u\n",
+			__func__, r->radar_type_idx, r->periodic_pulse_num, r->long_pulse_num, r->hw_pulse_num);
+
 	ieee80211_radar_detected(mphy->hw, NULL);
 	dev->hw_pattern++;
 }
--- a/mt7915/mcu.c
+++ b/mt7915/mcu.c
@@ -336,6 +336,12 @@ mt7915_mcu_rx_radar_detected(struct mt79
 	if (!mphy)
 		return;
 
+	dev_info(dev->mt76.dev, "%s: RADAR DETECTED r->rdd_idx=%d\n", __func__, r->rdd_idx);
+	dev_dbg(dev->mt76.dev, "%s: long_detected=%u constant_prf_detected=%u staggered_prf_detected=%u\n",
+			__func__, r->long_detected, r->constant_prf_detected, r->staggered_prf_detected);
+	dev_dbg(dev->mt76.dev, "%s: radar_type_idx=%u periodic_pulse_num=%u long_pulse_num=%u hw_pulse_num=%u\n",
+			__func__, r->radar_type_idx, r->periodic_pulse_num, r->long_pulse_num, r->hw_pulse_num);
+
 	if (r->rdd_idx == MT_RDD_IDX_BACKGROUND)
 		cfg80211_background_radar_event(mphy->hw->wiphy,
 						&dev->rdd2_chandef,
--- a/mt7996/mcu.c
+++ b/mt7996/mcu.c
@@ -549,6 +549,12 @@ mt7996_mcu_rx_radar_detected(struct mt79
 	if (!mphy)
 		goto err;
 
+	dev_info(dev->mt76.dev, "%s: RADAR DETECTED r->rdd_idx=%d\n", __func__, r->rdd_idx);
+	dev_dbg(dev->mt76.dev, "%s: long_detected=%u constant_prf_detected=%u staggered_prf_detected=%u\n",
+			__func__, r->long_detected, r->constant_prf_detected, r->staggered_prf_detected);
+	dev_dbg(dev->mt76.dev, "%s: radar_type_idx=%u periodic_pulse_num=%u long_pulse_num=%u hw_pulse_num=%u\n",
+			__func__, r->radar_type_idx, r->periodic_pulse_num, r->long_pulse_num, r->hw_pulse_num);
+
 	if (r->rdd_idx == MT_RDD_IDX_BACKGROUND) {
 		cfg80211_background_radar_event(mphy->hw->wiphy,
 						&dev->rdd2_chandef,
