--- a/mt7915/mcu.c
+++ b/mt7915/mcu.c
@@ -162,8 +162,19 @@ mt7915_mcu_parse_response(struct mt76_de
 	int ret = 0;
 
 	if (!skb) {
-		dev_err(mdev->dev, "Message %08x (seq %d) timeout\n",
-			cmd, seq);
+		const char* first = "Secondary";
+
+		if (!mdev->first_failed_mcu_cmd)
+			first = "Initial";
+
+		dev_err(mdev->dev, "MCU: %s Failure: Message %08x (cid %lx ext_cid: %lx seq %d) timeout. Last successful cmd: 0x%x\n",
+			first,
+			cmd, FIELD_GET(__MCU_CMD_FIELD_ID, cmd),
+			FIELD_GET(__MCU_CMD_FIELD_EXT_ID, cmd), seq,
+			mdev->last_successful_mcu_cmd);
+
+		if (!mdev->first_failed_mcu_cmd)
+			mdev->first_failed_mcu_cmd = cmd;
 
 		if (!test_and_set_bit(MT76_MCU_RESET, &dev->mphy.state)) {
 			dev->recovery.restart = true;
@@ -175,11 +186,25 @@ mt7915_mcu_parse_response(struct mt76_de
 		return -ETIMEDOUT;
 	}
 
+	mdev->last_successful_mcu_cmd = cmd;
+
+	if (mdev->first_failed_mcu_cmd) {
+		dev_err(mdev->dev, "MCU: First success after failure: Message %08x (cid %lx ext_cid: %lx seq %d)\n",
+			cmd, FIELD_GET(__MCU_CMD_FIELD_ID, cmd),
+			FIELD_GET(__MCU_CMD_FIELD_EXT_ID, cmd), seq);
+		mdev->first_failed_mcu_cmd = 0;
+	}
+
 	rxd = (struct mt76_connac2_mcu_rxd *)skb->data;
 	if (seq != rxd->seq &&
 	    !(rxd->eid == MCU_CMD_EXT_CID &&
 	      rxd->ext_eid == MCU_EXT_EVENT_WA_TX_STAT))
+	{
+		dev_dbg(mdev->dev, "ERROR: MCU: Sequence mismatch in response, seq: %d rxd->seq: %d cmd: %0x\n",
+			seq, rxd->seq, cmd);
+
 		return -EAGAIN;
+	}
 
 	if (cmd == MCU_CMD(PATCH_SEM_CONTROL)) {
 		skb_pull(skb, sizeof(*rxd) - 4);
--- a/mt76.h
+++ b/mt76.h
@@ -1051,6 +1051,9 @@ struct mt76_dev {
 	};
 
 	atomic_t bus_hung;
+
+	int first_failed_mcu_cmd;
+	int last_successful_mcu_cmd;
 };
 
 /* per-phy stats.  */
